app/controllers/user_badges_controller.rb

```diff
@@ -91,6 +94,20 @@ def destroy
     render json: success_json
   end
 
+  def favorite
+    params.require(:id)
+    user_badge = UserBadge.find(params[:id])
+    user_badges = user_badge.user.user_badges
+
+    if !user_badge.is_favorite && user_badges.where(is_favorite: true).count >= MAX_FAVORITES
+      render json: failed_json, status: 403
+    else
+      user_badge.update_attribute(:is_favorite, !user_badge.is_favorite)
+      UserBadge.update_featured_ranks!(user_badge.user_id)
+      render_serialized(user_badge, DetailedUserBadgeSerializer, root: "user_badge")
```

Comment:
Since you changed it to a symbol on line 78, it's best to keep it consistent here as well

```suggestion
      render_serialized(user_badge, DetailedUserBadgeSerializer, root: :user_badge)
```

Reply:
Oh whoops, I did not mean to modify that I must've accidentally format with auto-format on save. Thanks for picking that up!